### DISCUSSION TITLE
Support for multiple radio interfaces

### DISCUSSION THREAD

**Discussion by code8buster** [meshtastic/firmware#3831] [Ideas]
There is at least one board in the protobufs with two SPI Ebyte modules on board, selectable via CS/NSS pin. (disasterradio v3 DR_DEV)
Currently, there is no way to use both simultaneously, nor any logic in the Router to decide which module a packet should be sent to, or track which module a packet came from. 

thebentern and sachaw suggested I create a tracking issue. To enhance the project in this way would likely require changes to RadioInterface, Router, and a slew of other core modules.

**Comment by Krezalis** [comment]
It can be a very helpful functionality at repeater node 433/868. No need to use two ESP and an additional IC to route messages via UART.

**Comment by m-anish** [comment]
This would be a very nice feature to have. :)

**Comment by Ccz119** [comment]
I can think many uses for this !

**Comment by Afrikaaner** [comment]
This seems like a critical feature: specifically with two nodes, I can see this enabling infrastructure nodes to communicate with each other simultaneously to client nodes. Add a third radio, and now infrastructure nodes have the opportunity to choose to communicate over either a high or low bandwidth connection with each other (a la 10/100/1000 mbps LAN).

**Comment by Ryu945** [comment]
It can also be useful to run a 2.4 Ghz and 900 MHz node at the same time.  Thus you can use the 2.4 GHz for shorter range and higher bandwidth applications while not losing the long range communication of 900 MHz.

**Comment by Afrikaaner** [comment]
Serious question, what benefit would short range 2.4ghz provide to Meshtastic? I might be lacking imagination, but clients wouldn’t be able to connect without new radios, and infrastructure nodes would likely be out of range of each other to communicate using 2.4. 

I’m imagining taking advantage of more of the spectrum we have available rather than just what Long_Fast provides. Maybe the client default becomes Medium_Fast to reduce congestion in heavy regions, and infrastructure nodes use Long_Fast and Long_Slow to reach further in fewer hops. Or maybe client default remains Long_Fast and infrastructure nodes also use Long_Slow to reach some further nodes. Or maybe they leverage multiple modes to bond bandwidth.

**Reply by Ryu945** [reply]
While it takes nodes to be closer together, your get so much more bandwidth with 2.4 Ghz that new potential features like voice communication can be possible.

I did a quick estimate with someone and came to the conclusion that 2.4 Ghz running Long Slow has roughly the same range as 900 MHz between Medium Fast and Medium Slow in a desert environment.   Probably lower in real life.  Applications where you use Short with 900 MHz would probably work better with 2.4 Ghz in its longest range configuration.

I also suspect that 2.4 GHz is not fully optimized as well as 900 Mhz is in Meshtastic.  Can more bandwidth be used to increase the range since you get much more bandwidth with it?

**Comment by ChaoticLeah** [comment]
This would also be great for the fact that I live in an area where some people use LONG_FAST and some people use MEDIUM_FAST. Allowing us to connect and configure multiple LoRa radios would be great to link them and also allow me to see and communicate with people on both settings.

**Reply by Afrikaaner** [reply]
Allowing a node to serve as an intermediary between different bands is an interesting idea. I could also see a client device with multiple radios using one radio to find the fastest reliable option before switching the primary radio over to it. Or using two bands simultaneously (essentially bonding them together) for additional bandwidth. Or even using both on the same frequency to do some sort of beam-forming like wireless routers use to focus signal and extend range or even allow a faster bands to be used where they would have been unable to before (presumably need to be careful not to exceed regional radiation limits though). 

The takeaway is the value of multiple radios is difficult to overstate. There are tons of potential use cases.

**Comment by Andrik45719** [comment]
**The world’s first dual-band Meshtastic!**

HT-CT62 + LoRa RA-01

I managed to initialize two modules over SPI — they’re even transmitting and receiving something, but only on the same frequency.
Now I need help from the community to add support for separate presets for the modems and to figure out the routing.

![dualband1](https://github.com/user-attachments/assets/540bb793-ff20-447e-ab17-276795f31c0a)
![dualband2](https://github.com/user-attachments/assets/da468320-6f9c-4bba-b0d9-aef528dff9e0)

```
DEBUG | ??:??:?? 1 RF95Interface(cs=2, irq=19, rst=9, busy=18)
INFO  | ??:??:?? 1 RF95 init success
DEBUG | ??:??:?? 1 SX126xInterface(cs=8, irq=3, rst=5, busy=4)
INFO  | ??:??:?? 2 SX1262 init success

DEBUG | 09:57:05 31 [RadioIf] Started Tx (id=0x9b43bed3 fr=0xe3137778 to=0xffffffff, transport = 0, WantAck=0, HopLim=5 Ch=0x62 encrypted len=91 rxtime=1762329424 hopStart=5 relay=0x78 pri
DEBUG | 09:57:07 33 [RadioIf] Lora RX (id=0x9b43bed3 fr=0xe3137778 to=0xffffffff, transport = 0, WantAck=0, HopLim=4 Ch=0x62 encrypted len=91 rxSNR=5.25 rxRSSI=-50 hopStart=5 relay=0xfc)
```
